### PR TITLE
Trac2869 fixes

### PR DESCRIPTION
**Pull request by olgamelnichuk**
@alf239 - please, review

### CODE REVIEW

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java [hunk 1]
```diff
@@ -195,20 +195,18 @@ public class ApiQueryRequestHandler extends AbstractRestRequestHandler implement
...
-                                                            experiment,
-                                                            geneIds,
-                                                            pathToNetCDFProxy,
-                                                            conditions,
-                                                            statFilter,
-                                                            queryResultSortOrder,
-                                                            queryStart,
-                                                            queryRows);
-                                        }
+                                    BestDesignElementsResult geneResults = BestDesignElementsResult.empty();
```
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java]: Make it final and not initialised, make an if statement with two branches. Or, perhaps, just use ternary statement.

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java [hunk 2]
```diff
@@ -195,20 +195,18 @@ public class ApiQueryRequestHandler extends AbstractRestRequestHandler implement
                                     NetCDFDescriptor pathToNetCDFProxy = atlasNetCDFDAO.getNetCdfFile(experiment.getAccession(), netCDFProxyPredicate);
```
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/ApiQueryRequestHandler.java:195]: rename to ncdf (yep, unrelated, but would be nice)

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java [hunk 3]
```diff
@@ -323,6 +321,7 @@ public class ExperimentResultAdapter {
         @RestOut(name = "designElementAccession")
         public String getDesignElementAccession() {
             String adAcc = getArrayDesignAccession();
+            //TODO: if adAcc == null => NPE?
```
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java:324]: you know what I want to say :)

Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java:324]: at least, add a unit test which would either fail or prove we have no problem here

Comment (olgamelnichuk) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java:324]: I prefer to postpone with this one; until we move out all non-api code from ExperimentResultAdapter.
The probability of NPE here is very small.. like existing experiment without an array design. But the code allows this - so it should be rewritten.

**Comment by alf** [comment]
@olgamelnichuk Please use @alf239 in the future as I don't really have much interest in reviewing your changes.

File: atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java (hunk 3 above)
Comment (alf239) [inline atlas-web/src/main/java/uk/ac/ebi/gxa/requesthandlers/api/result/ExperimentResultAdapter.java:324]: okay

**Comment by alf239** [comment]
reviewed, ok.

it would be nice to make color selection more sticky (e.g. change selection from 124 to 1245 to 125 to 1256 to 12456 and make sure the colour of 4 is always the same)—but it's not too important

**Comment by olgamelnichuk** [comment]
@alf - Sorry, was my mistake; too many Alfs :))

**Pull request merged**